pkg/controller/operators/olm/operator.go

```diff
@@ -452,55 +511,79 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
 		"namespace": clusterServiceVersion.GetNamespace(),
 		"phase":     clusterServiceVersion.Status.Phase,
 	})
-	logger.Info("syncing CSV")
+	logger.Debug("syncing CSV")
 
 	outCSV, syncError := a.transitionCSVState(*clusterServiceVersion)
 
-	// no changes in status, don't update
-	if outCSV.Status.LastUpdateTime == clusterServiceVersion.Status.LastUpdateTime &&
-		outCSV.Status.Phase == clusterServiceVersion.Status.Phase &&
-		outCSV.Status.Reason == clusterServiceVersion.Status.Reason &&
-		outCSV.Status.Message == clusterServiceVersion.Status.Message {
+	if outCSV == nil {
 		return
 	}
 
-	// Update CSV with status of transition. Log errors if we can't write them to the status.
-	updatedCSV, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(clusterServiceVersion.GetNamespace()).UpdateStatus(outCSV)
-	if err != nil {
-		updateErr := errors.New("error updating ClusterServiceVersion status: " + err.Error())
-		if syncError == nil {
-			logger.Info(updateErr)
-			return updateErr
+	// status changed, update CSV
+	if !(outCSV.Status.LastUpdateTime == clusterServiceVersion.Status.LastUpdateTime &&
+		outCSV.Status.Phase == clusterServiceVersion.Status.Phase &&
+		outCSV.Status.Reason == clusterServiceVersion.Status.Reason &&
+		outCSV.Status.Message == clusterServiceVersion.Status.Message) {
+
+		// Update CSV with status of transition. Log errors if we can't write them to the status.
+		_, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(outCSV.GetNamespace()).UpdateStatus(outCSV)
+		if err != nil {
+			updateErr := errors.New("error updating ClusterServiceVersion status: " + err.Error())
+			if syncError == nil {
+				logger.Info(updateErr)
+				return updateErr
+			}
+			syncError = fmt.Errorf("error transitioning ClusterServiceVersion: %s and error updating CSV status: %s", syncError, updateErr)
 		}
-		syncError = fmt.Errorf("error transitioning ClusterServiceVersion: %s and error updating CSV status: %s", syncError, updateErr)
 	}
 
-	operatorGroup := a.operatorGroupForActiveCSV(logger, updatedCSV)
-	if operatorGroup == nil {
-		logger.WithField("reason", "no operatorgroup found for active CSV").Info("skipping CSV resource copy to target namespaces")
+	if outCSV.IsUncopiable() {
```

Comment:
I think you also want this check in `syncCopyCSV` since it's also being fed by an informer.